enterprise/cypher/compiled-expressions/src/test/scala/org/neo4j/cypher/internal/runtime/compiled/expressions/CodeGenerationTest.scala

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright (c) 2002-2018 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j Enterprise Edition. The included source
+ * code can be redistributed and/or modified under the terms of the
+ * GNU AFFERO GENERAL PUBLIC LICENSE Version 3
+ * (http://www.fsf.org/licensing/licenses/agpl-3.0.html) with the
+ * Commons Clause, as found in the associated LICENSE.txt file.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * Neo4j object code can be licensed independently from the source
+ * under separate terms from the AGPL. Inquiries can be directed to:
+ * licensing@neo4j.com
+ *
+ * More information is also available at:
+ * https://neo4j.com/licensing/
+ */
+package org.neo4j.cypher.internal.runtime.compiled.expressions
+
+import java.lang.Math.PI
+import java.util.concurrent.ThreadLocalRandom
+
+import org.neo4j.cypher.internal.runtime.interpreted.ExecutionContext
+import org.neo4j.internal.kernel.api.Transaction
+import org.neo4j.values.storable.Values._
+import org.neo4j.values.storable.{DoubleValue, Values}
+import org.neo4j.values.virtual.VirtualValues
+import org.neo4j.values.virtual.VirtualValues.{EMPTY_MAP, list, map}
+import org.opencypher.v9_0.ast.AstConstructionTestSupport
+import org.opencypher.v9_0.expressions._
+import org.opencypher.v9_0.util.symbols
+import org.opencypher.v9_0.util.test_helpers.CypherFunSuite
+
+class CodeGenerationTest extends CypherFunSuite with AstConstructionTestSupport {
+
+  private val tx = mock[Transaction]
+  private val ctx = mock[ExecutionContext]
+  private val random = ThreadLocalRandom.current()
+
+  test("round function") {
+    compile(function("round", literalFloat(PI))).compute(ctx, tx, EMPTY_MAP) should equal(doubleValue(3.0))
+    compile(function("round", noValue)).compute(ctx, tx, EMPTY_MAP) should equal(NO_VALUE)
+  }
+  test("rand function") {
+    // Given
+    val expression = function("rand")
+
+    // When
+    val compiled = compile(expression)
+
+    // Then
+    val value = compiled.compute(ctx, tx, EMPTY_MAP).asInstanceOf[DoubleValue].doubleValue()
+    value should (be > 0.0 and be <= 1.0)
```

Comment:
Isn't it `>= 0.0`? Although I guess we will not hit that...

Reply:
yes I was confused